src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+
 public class SalaryInfo {
+    private static final String DATA_RECORD_SPLIT_REGEX = " ";
+    private static final int SALARY_OFFSET = 0;
+    private static final int DATE_COLUMN = 0;
+    private static final int NAME_COLUMN = 1;
+    private static final int WORKING_HOURS_COLUMN = 2;
+    private static final int SALARY_RATE_COLUMN = 3;
+    private int[] totalSalary = new int[] {};
+    private String dataPattern = "dd.MM.yyyy";
+    private DateTimeFormatter simpleDateFormat = DateTimeFormatter.ofPattern(dataPattern);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        totalSalary = new int[names.length];
+        for (int i = 0; i < totalSalary.length; i++) {
+            totalSalary[i] = SALARY_OFFSET;
+        }
+        for (String dataRecord : data) {
+            String[] splitRecord = dataRecord.split(DATA_RECORD_SPLIT_REGEX);
+            String requestDate = splitRecord[DATE_COLUMN];
+            if (checkIfDateIsInRange(dateFrom, dateTo, requestDate)) {
+                int salary = Integer.parseInt(splitRecord[WORKING_HOURS_COLUMN])
+                        * Integer.parseInt(splitRecord[SALARY_RATE_COLUMN]);
+                totalSalary[indexOf(names,splitRecord[NAME_COLUMN])] += salary;
```

Comment:
```suggestion
                totalSalary[indexOf(names, splitRecord[NAME_COLUMN])] += salary;
```
use IDE's auto formatter before committing